### PR TITLE
Add generative search to GRPC

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

Adds generative search to GRPC

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/grpc/parse_request.go [hunk 1]
```diff
@@ -234,6 +242,20 @@ func searchParamsFromProto(req *pb.SearchRequest, scheme schema.Schema) (dto.Get
...
 
+func extractGenerative(req *pb.SearchRequest) *generate.Params {
+	generative := generate.Params{}
+	if req.Generative.SingleResponsePrompt != "" {
+		generative.Prompt = &req.Generative.SingleResponsePrompt
+	}
+	if req.Generative.GroupedResponseTask != "" {
+		generative.Task = &req.Generative.GroupedResponseTask
+	}
+	if req.Generative.GroupedProperties != nil && len(req.Generative.GroupedProperties) > 0 {
```
Comment (aliszka) [inline adapters/handlers/grpc/parse_request.go:253]: I believe `len` works well with nil slices, so 1st check could be removed

File: adapters/handlers/grpc/prepare_reply.go [hunk 2]
```diff
@@ -60,27 +68,60 @@ func searchResultsToProto(res []any, start time.Time, searchParams dto.GetParams
...
+		return additionalProps, "", nil
 	}
 
 	additionalPropertiesMap := asMap["_additional"].(map[string]interface{})
+	generativeGroupResults := ""
+	// id is part of the _additional map in case of generative search - don't aks me why
+	if generativeSearchEnabled {
+		idRaw, ok := additionalPropertiesMap["id"]
+		if !ok {
+			return nil, generativeGroupResults, errors.Wrap(err, "get id generative")
```
Comment (aliszka) [inline adapters/handlers/grpc/prepare_reply.go:99]: although `generativeGroupResults` is empty at this point I would explicitly return `""` for clarity

File: adapters/handlers/grpc/prepare_reply.go [hunk 3]
```diff
@@ -60,27 +68,60 @@ func searchResultsToProto(res []any, start time.Time, searchParams dto.GetParams
...
+		}
+
+		idStrfmt, ok := idRaw.(strfmt.UUID)
+		if !ok {
+			return nil, generativeGroupResults, errors.Wrap(err, "format id generative")
```
Comment (aliszka) [inline adapters/handlers/grpc/prepare_reply.go:104]: although `generativeGroupResults` is empty at this point I would explicitly return `""` for clarity

File: adapters/handlers/grpc/prepare_reply.go [hunk 4]
```diff
@@ -60,27 +68,60 @@ func searchResultsToProto(res []any, start time.Time, searchParams dto.GetParams
...
 	additionalPropertiesMap := asMap["_additional"].(map[string]interface{})
+	generativeGroupResults := ""
+	// id is part of the _additional map in case of generative search - don't aks me why
+	if generativeSearchEnabled {
+		idRaw, ok := additionalPropertiesMap["id"]
+		if !ok {
+			return nil, generativeGroupResults, errors.Wrap(err, "get id generative")
+		}
+
+		idStrfmt, ok := idRaw.(strfmt.UUID)
```
Comment (aliszka) [inline adapters/handlers/grpc/prepare_reply.go:102]: is it needed? `idStrfmt` is casted to string later anyway

File: adapters/handlers/grpc/prepare_reply.go [hunk 5]
```diff
@@ -60,27 +68,60 @@ func searchResultsToProto(res []any, start time.Time, searchParams dto.GetParams
...
+		if !ok {
+			return nil, generativeGroupResults, errors.Wrap(err, "format id generative")
+		}
+		additionalProps.Id = idStrfmt.String()
+
+		generate, ok := additionalPropertiesMap["generate"]
+		if ok { // does not always have content, for example with grouped results only the first object has an entry
+			generateFmt, ok := generate.(*models.GenerateResult)
+			if !ok {
+				return nil, generativeGroupResults, errors.Wrap(err, "cast generative result")
```
Comment (aliszka) [inline adapters/handlers/grpc/prepare_reply.go:112]: although `generativeGroupResults` is empty at this point I would explicitly return `""` for clarity

**Review by aliszka — APPROVED** [review]

File: adapters/handlers/grpc/prepare_reply.go (hunk 5 above)
Comment (dirkkul) [reply adapters/handlers/grpc/prepare_reply.go:112]: Not really needed, but this gurantees that the ID is actually in the right format. I'd prefer to leave it in

File: adapters/handlers/grpc/prepare_reply.go (hunk 4 above)
Comment (dirkkul) [reply adapters/handlers/grpc/prepare_reply.go:102]: Intuitive I would want the same object to come back, with a new attribute for the generative output. It already has metadata like distances anyway.

**Review by aliszka — APPROVED** [review]

**Pull request merged**